app/controllers/shared/api/auth.php

```diff
@@ -46,15 +48,17 @@
             }
         }
 
-        $route = $utopia->match($request);
-
-        $isPrivilegedUser = Auth::isPrivilegedUser(Authorization::getRoles());
-        $isAppUser = Auth::isAppUser(Authorization::getRoles());
+        $isPrivilegedUser = Auth::isPrivilegedUser($authorization->getRoles());
+        $isAppUser = Auth::isAppUser($authorization->getRoles());
 
         if ($isAppUser || $isPrivilegedUser) { // Skip limits for app and console devs
             return;
         }
 
+        if (str_contains($route->getPath(), '/v1/graphql')) { // Skip for graphQL recursive call
```

Comment:
this check doesnt feel safe. Maybe use `route->getLabel('sdk.namespace') === 'graphql')`